ixmp4/data/db/optimization/table/model.py

```diff
@@ -22,10 +24,12 @@ class Table(base.BaseModel):
     columns: types.Mapped[list["Column"]] = db.relationship()
     data: types.JsonDict = db.Column(db.JsonType, nullable=False, default={})
 
+    # TODO: should we pass self to validate_data to raise more specific errors?
+
     @validates("data")
     def validate_data(self, key, data: dict[str, Any]):
         return utils.validate_data(
-            key=key,
+            host=self,
```

Comment:
Passing the whole instance at the moment to access both `.DataInvalid` and `__str__`, but if we change the way we raise errros in `utils.validate_data()`, we might also be able to just pass `__str__`.